### ISSUE TITLE
aws.ssm.getParameter doesn't return value

### ISSUE THREAD

**Issue by sebastianfeduniak** [pulumi/pulumi-aws#4011]
### What happened?

I use the below code:

```
aws.ssm.getParameter({ name: 'jwt-signing-secret-key' }).then(
    () => aws.ssm.Parameter.get('jwt-signing-secret-key', 'jwt-signing-secret-key:1'),
    () =>
      new aws.ssm.Parameter('jwt-signing-secret-key', {
        name: 'jwt-signing-secret-key',
        type: 'SecureString',
        value: generateJwtSecretKey(),
      })
  );
```

The parameter does exist in the AWS.
And the code works good in `"@pulumi/aws": "6.0.2"` but doesn't work in the latest version `6.38`.
In the latest version it throws error:
```
  aws:ssm:Parameter (jwt-signing-secret-key):
    error: Preview failed: resource 'jwt-signing-secret-key:1' does not exist
```

### Example

aws.ssm.getParameter({ name: 'jwt-signing-secret-key' }).then(
    () => aws.ssm.Parameter.get('jwt-signing-secret-key', 'jwt-signing-secret-key:1'),
    () =>
      new aws.ssm.Parameter('jwt-signing-secret-key', {
        name: 'jwt-signing-secret-key',
        type: 'SecureString',
        value: generateJwtSecretKey(),
      })
  );

### Output of `pulumi about`

CLI
Version      3.117.0
Go Version   go1.22.3
Go Compiler  gc

Plugins
KIND      NAME             VERSION
resource  aws              6.38.0
resource  awsx             2.11.0
resource  command          0.7.1
resource  confluentcloud   1.46.0
resource  datadog          4.23.0
resource  docker           4.5.4
resource  docker           3.6.1
resource  docker           3.6.1
resource  eventstorecloud  0.2.10
language  nodejs           unknown
resource  postgresql       3.6.0
resource  random           4.13.2

Host
OS       darwin
Version  12.7
Arch     x86_64

Dependencies:
NAME                                                VERSION
eslint-config-prettier                              8.10.0
randomstring                                        1.3.0
@tsconfig/node14                                    1.0.3
@types/node                                         18.19.33
@typescript-eslint/eslint-plugin                    5.62.0
@typescript-eslint/parser                           5.62.0
eslint-config-airbnb                                19.0.4
eslint                                              8.57.0
jsonwebtoken                                        9.0.2
@pulumi/pulumi                                      3.86.0
eslint-config-airbnb-typescript                     17.1.0
eslint-plugin-prettier                              4.2.1
prettier                                            2.8.8

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by sebastianfeduniak** [comment]
Found out that it is working in `6.27.0` version. So something happened between `6.27.0` and `6.38`.

**Comment by sebastianfeduniak** [comment]
Thanks for the update!

**Comment by flostadler** [comment]
This was fixed in #4014. A new release  is currently being created

**Comment by sebastianfeduniak** [comment]
Thanks for handling it so quickly!
